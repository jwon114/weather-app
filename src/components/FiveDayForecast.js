import React from 'react';
import PropTypes from 'prop-types';
import DailyForecast from './DailyForecast';
import { Transition } from 'react-spring/renderprops';

const FiveDayForecast = (props) => (
  <div className="five-day-forecast">
    {props.data.length === 0 && 
      <div className="five-day-forecast__data-not-available">No weather forecast data available.</div>}
    {props.data.length > 0 &&
      <Transition
        items={props.data} 
        keys={item => item.dt}
        from={{ opacity: 0 }}
        enter={{ opacity: 1 }}
        reset={props.animate}
        config={{ duration: 500 }}
        trail={200} >
          {item => styleProp =>
            <DailyForecast
              style={styleProp}
              day={item.dt}
              temperature={item.main.temp}
              iconCode={item.weather[0].icon}
              forecast={item.weather[0].description} />}
      </Transition>}
  </div>
);

FiveDayForecast.propTypes = {
  data: PropTypes.array.isRequired,
  animate: PropTypes.bool.isRequired
};

export default FiveDayForecast;