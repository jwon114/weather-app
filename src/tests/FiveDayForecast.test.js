import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import FiveDayForecast from '../components/FiveDayForecast';

test('should render Forecast correctly', () => {
  const root = create(<FiveDayForecast 
                            data={[]}
                            animate={false}
                            loading={false} />);
  
  expect(root.toJSON()).toMatchSnapshot();
});

test('should render the the next 5 days', () => {
  const testData = [
    {
      dt: 1587985200, // Monday
      main: { temp: 18.69 },
      weather: [{ description: 'light rain', icon: '10d' }]
    },
    {
      dt: 1588071600, // Tuesday
      main: { temp: 9.17 },
      weather: [{ description: 'clear sky', icon: '01d' }]
    },
    {
      dt: 1588158000, // Wednesday
      main: { temp: 10.85 },
      weather: [{ description: 'cloudy', icon: '02d' }]
    },
    {
      dt: 1588244400, // Thursday
      main: { temp: 15.20 },
      weather: [{ description: 'snow', icon: '13d' }]
    },
    {
      dt: 1588330800, // Friday
      main: { temp: 11.37 },
      weather: [{ description: 'heavy rain', icon: '09d' }]
    }
  ];
  const { container, getByText } = render(<FiveDayForecast 
                                            data={testData} 
                                            animate={false}
                                            loading={false} />);
  
  expect(container.firstChild.children.length).toBe(5);
  expect(getByText(/mon/i)).toBeInTheDocument();
  expect(getByText(/tue/i)).toBeInTheDocument();
  expect(getByText(/wed/i)).toBeInTheDocument();
  expect(getByText(/thu/i)).toBeInTheDocument();
  expect(getByText(/fri/i)).toBeInTheDocument();
});
