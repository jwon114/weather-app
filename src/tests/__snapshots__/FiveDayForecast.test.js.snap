// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Forecast correctly 1`] = `
<div
  className="five-day-forecast"
>
  <div
    className="five-day-forecast__data-not-available"
  >
    No weather forecast data available.
  </div>
</div>
`;
